### ISSUE TITLE
[New Feature] Custom Building Foundations

### ISSUE THREAD

**Issue by Crimsonum** [Vinifera-Developers/Vinifera#104]
### Description:
Building foundations (`Foundation=`) are limited to a hardcoded list of possible shapes and sizes. It would greatly benefit modders to have more foundation options.

For reference, Ares allows [fully customizable building foundations](http://ares-developers.github.io/Ares-docs/new/buildings/custombuildingfoundations.html).

### Possible Implementation:
Allow modders to define custom foundations for buildings in art.ini. See Ares' system for reference. However, unlike Ares, I would make the system based on named foundation types, i.e. `Foundation=<foundation_name>`, so different building types can share the same custom foundation. These foundations could be defined elsewhere in art.ini, like infantry animation sequences are.

**Comment by CCHyper** [comment]
I agree that a named system would be best for defining new size types. For example;

`ART.INI`
```
[Foundations]
0=4x9

[Foundation_4x9]
Width=4
Height=9
OccupyCell1=0,0
OverlapCell1=-1,-1
etc...

[GACNST]
Foundation=4x9
```
